nice-to-have-patterns.md

```diff
@@ -0,0 +1,152 @@
+    - Git
+      - Commits should be descriptive, and explain changes contained in them
+      - PRs should be kept small to make for an easier review process
+    - React
+      - React components should be as dumb as possible
+        - Logic should be abstracted away to redux/containers/util functions/api/etc
+          - Good:
+            - TBD
+          - Bad:
+            - TBD
+      - Avoid using index.js for React component names
+        - This will help avoid having 10+ index.js tabs open in the editor
+        - It will also help make the components searchable using fuzzy search
+    - Code
+      - Use meaningful/descriptive names
+        - Bad:
+          - ```javascript
+             ps.map(p => {})
+        - Good:
+          - ```javascript
+             products.map(product => {})
+      - Functions/classes should ideally have one responsibility.
+        - Single responsibility functions are easier to reuse/test/read/refactor
+        - Use functional decomposition to abstract away logic
+        - Bad:
+          - ```javascript
+              function renderRedProducts(products) {
+                  const filteredProducts = products.map(...)
+                      .filter(...)
+                      .someOtherFunction(id)
+                  .
+                  .
+                  .
+                  .
+                   // Some more data mapping/logic ^
+                  return filteredProducts.map(product => <SomeElement />)
+              }
+        - Good:
+          - ```javascript
+            - function getRedProducts(products) {
+                 const filteredProducts =  products.map(...)
+                      .filter(...)
+                      .someOtherFunction(id)
+                  .
+                  .
+                  .
+                  .
+                   // Some more data mapping/logic ^
+
+                 return filteredProducts
+              }
+
+              function renderRedProducts(products) {
+                  const productsToRender = getRedProducts(products)
+                  return productsToRender.map(product => <SomeElement />)
+              }
+
+      - Encapsulate boolean expressions for readability
+        - Bad:
+          - ```javascript
+            if (someArray.length > 2 && someArray[0].id === "someId") {
+                ...
+            }
+        - Good:
+          - ```javascript
+            const isSomeCondition = someArray.length > 2 && someArray[0].id === "someId";
+
+            if (isSomeCondition) {
+                ...
+            }
+      - Prefer using native JS features over outside sources/lodash
+        - Modern JS Array/Number/String prototype functions are faster and generally more reliable
+        - This is specially true on the mobile side since we don't have to worry about browser compatibility 
+        - On flip side no need to reinvent the wheel from scratch if an outside library can save time/provide an already tested/stable solution
```

Comment:
what does this mean? I think we should prefer away from outside libraries in general. If outside library is to be used/suggested, should vote as a team first

Reply:
I agree with what you're saying. The case that I had in mind for the last bullet was trying to create something like React from scratch, not a small utils library. Any ideas on how to modify it to communicate this better?

I'll add the part about voting as a team.